bigquery/params.go

```diff
@@ -380,6 +381,12 @@ func paramType(t reflect.Type, v reflect.Value) (*bq.QueryParameterType, error)
 			return nil, err
 		}
 		for _, f := range fields {
+			prefixes := []string{"*", "[]"} // check pointer and arrays
+			for _, prefix := range prefixes {
+				if strings.TrimPrefix(t.String(), prefix) == strings.TrimPrefix(f.Type.String(), prefix) {
+					return nil, fmt.Errorf("bigquery: Go type %s cannot be represented as a parameter due to a cycle detected", t)
```

Comment:
Do we want to be clearer that this is a recursive type problem?  I worry that only mentioning a cycle is insufficient?